scipy/optimize/_linprog.py

```diff
@@ -587,7 +591,11 @@ def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
     if x0 is not None and meth != "revised simplex":
         warning_message = "x0 is used only when method is 'revised simplex'. "
         warn(warning_message, OptimizeWarning)
-
+        
+    if 'simplex' in meth:
+        A_eq = A_eq.todense()
+        A_ub = A_ub.todense()
```

Comment:
This will _only_ work for sparse input, right? I suggest following `_LPProblem` or `_parse_linprog` - I can't remember which - to find out where the input validation is. Try to find other input validation on `A_eq` and `A_ub` and put this there, and only convert `todense` if `issparse`.

Reply:
Just to clarify, the issue here is that, the way I've written it, if someone choose a simplex method, but submits a dense matrix/array, then this will raise an exception. So, it should check first that it needs to convert, and then only do so if `issparse`. We want the user to be able to submit either sparse or dense without making `_linprog()` angry.

`_parse_linprog()` calls `check_sparse_inputs()`, which at one point calls `sparse.issparse(A_eq)` to see if the inputs are sparse matrices. I think it would make sense to just call `sparse.isspsarse` directly because `check_sparse_inputs()` will throw an exception if the the sparsity constraint and the method are not compatible. 

I'll push changes shortly to reflecting the above.

Reply:
> Just to clarify...

Exactly. 

> I think it would make sense to just call...

I'm not sure about that. It would be unusual for the code to convert the matrices to dense _and then_ check whether they are sparse, right? They can't be sparse if our code has already changed them to dense.

Instead, we'd want to _change_ that code in `_check_sparse_inputs` if it is no longer what we want to do. 

Let's hold off on changes while we wait back for @mckib2's opinion, though. I had forgotten that this check was in there, and I didn't realize that the code currently raises an exception for sparse input. Producing an error and directing users to use one of the HiGHS methods in this case might be better thing than converting to dense.